File: apps/Offline-AI-Buddy/android/app/src/main/java/com/americangroupllc/offlineaibuddy/settings/SettingsScreen.kt
package com.americangroupllc.offlineaibuddy.settings

import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.material3.ListItem
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp

@Composable
fun SettingsScreen() {
    LazyColumn(modifier = Modifier.fillMaxSize().padding(16.dp)) {
        item { ListItem(headlineContent = { Text("Default language") }) }
        item { ListItem(headlineContent = { Text("Voice settings") }) }
        item { ListItem(headlineContent = { Text("Wi-Fi-only model downloads") }, trailingContent = { Text("On") }) }
        item { ListItem(headlineContent = { Text("Theme") }, supportingContent = { Text("System") }) }
        item { ListItem(headlineContent = { Text("Premium / Subscribe / Restore") }) }
        item { ListItem(headlineContent = { Text("Erase all chats") }) }
        item { ListItem(headlineContent = { Text("Delete model") }) }
        item {
            ListItem(
                headlineContent = { Text("Privacy") },
                supportingContent = { Text("Offline AI Buddy does not send any data to us.") },
            )
        }
    }
}

File: apps/Offline-AI-Buddy/ios/OfflineAIBuddy/Features/Settings/SettingsScreen.swift
import SwiftUI
import BuddyAICore

struct SettingsScreen: View {
    @EnvironmentObject private var profilesModel: ProfilesModel
    @EnvironmentObject private var llama: LlamaService
    @AppStorage("settings.wifionly") private var wifiOnlyDownload: Bool = true
    @AppStorage("settings.tts.speed") private var ttsSpeed: Double = 0.5
    @AppStorage("settings.theme") private var theme: String = "system"

    var body: some View {
        NavigationStack {
            Form {
                Section("Language") {
                    Picker("Default language", selection: $profilesModel.defaultLanguage) {
                        ForEach(Language.allCases, id: \.self) { Text($0.displayName).tag($0) }
                    }
                }
                Section("Voice") {
                    HStack { Text("TTS speed"); Slider(value: $ttsSpeed, in: 0.2...1.0) }
                }
                Section("Connectivity") {
                    Toggle("Wi-Fi-only model downloads", isOn: $wifiOnlyDownload)
                }
                Section("Appearance") {
                    Picker("Theme", selection: $theme) {
                        Text("System").tag("system")
                        Text("Light").tag("light")
                        Text("Dark").tag("dark")
                    }
                }
                Section("Premium") {
                    NavigationLink("Subscribe / Restore") { SubscriptionScreen() }
                }
                Section("Storage") {
                    Button("Erase all chats", role: .destructive) {
                        try? ChatHistoryStore().eraseAll()
                    }
                    Button("Delete model", role: .destructive) {
                        try? llama.store.remove(named: "\(llama.manifest.name).gguf")
                    }
                }
                Section("Privacy") {
                    Text("Offline AI Buddy does not send any data to us.")
                        .font(.footnote).foregroundStyle(.secondary)
                }
            }
            .navigationTitle("Settings")
        }
    }
}
